subprojects/core/src/main/java/org/gradle/api/internal/changedetection/changes/NoHistoryArtifactState.java

```diff
@@ -80,4 +82,9 @@ public FileCollection getOutputFiles() {
     public OverlappingOutputs getOverlappingOutputs() {
         return null;
     }
+
+    @Override
+    public Set<File> getPreviousOutputs() {
+        return null;
+    }
```

Comment:
I removed the method. Do you want me to change the return value for `getOutputFiles` to `Optional<Set<File>>`? It feels somewhat wrong to use `Optional` of a collection, given that I could just return an empty collection. But we use an empty collection to mean something different - that we have no output files as opposed to having no history.